### DISCUSSION TITLE
I have problem with C++ github actions CodeQL

### DISCUSSION THREAD

**Discussion by turbosmurfen** [community/community#25779] [Actions]
<p>I’m new to use actions, but got a notice about that I should use this. Some how I get errors when using MSBuild for C++. Because I have an old version or something. Everything happened after I switched to Visual Studio 2022. Would be really nice if someone could help me out.</p>
<p>My current setup is this:</p>
<pre><code class="lang-auto">name: CodeQL

on:
  push:
    branches: [main]
  pull_request:
    types: [opened, synchronize]
    branches: [main]
  schedule:
    - cron: '18 20 * * 1'

env:
  # Path to the solution file relative to the root of the project.
  SOLUTION_FILE_PATH: spoton.sln

jobs:
  Analyze:
    runs-on: windows-latest

    strategy:
      fail-fast: false

    steps:
      - name: Checkout
        uses: actions/checkout@v2
        with:
          submodules: recursive

      # Initializes the CodeQL tools for scanning.
      - name: Initialize CodeQL
        uses: github/codeql-action/init@v1
        with:
          languages: cpp

      - name: Add MSBuild to PATH
        uses: microsoft/setup-msbuild@v1

      - name: Build debuggable x86 plugin DLL
        working-directory: ${{ github.workspace }}
        run: msbuild /m /p:Configuration="Debug" /p:Platform="x86" ${{ env.SOLUTION_FILE_PATH }}

      - name: Perform CodeQL Analysis
        uses: github/codeql-action/analyze@v1
</code></pre>
<p>And the error I get is something about platform toolset which I have no clue what to do with.<br>
Here is the whole error:</p>
<pre><code class="lang-auto">C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(439,5): error MSB8020: The build tools for v143 (Platform Toolset = 'v143') cannot be found. To build using the v143 build tools, please install v143 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution"
</code></pre>

**Comment by adityasharad** [comment]
<p>Hello. Thanks for informing us and sharing your workflow file.</p>
<p>To help us understand whether this is a CodeQL compatibility problem or a problem with your build, could you please let us know whether your build workflow succeeds <em>without</em> the two <code>github/codeql-action</code> steps?</p>

**Comment by adityasharad** [comment, marked as answer]
<p>I see this question is a duplicate of <a href="https://github.com/github/codeql/discussions/7162" class="inline-onebox">New to MSBuild and CodeQL for my C++ Project. (build tools v143 cannot be found) · Discussion #7162 · github/codeql · GitHub</a>. We’ll continue the discussion there.</p>
